core/os/thread_safe.h

```diff
@@ -31,8 +31,6 @@
 #ifndef THREAD_SAFE_H
 #define THREAD_SAFE_H
 
-#include "core/os/mutex.h"
```

Comment:
This was also a wrong suggestion from IWYU, as it's used just below in macros. Apparently it doesn't support macros, so we need to be cautious there too when reviewing changes.